libraries/CurieBLE/src/internal/BLECallbacks.cpp

```diff
@@ -77,7 +77,13 @@ ssize_t profile_longwrite_process(struct bt_conn *conn,
                                      const void *buf, uint16_t len,
                                      uint16_t offset)
 {
-    BLECharacteristicImp *blecharacteritic = (BLECharacteristicImp*)attr->user_data;
+    BLEAttribute *bleattr = (BLEAttribute *)attr->user_data;
```

Comment:
@sgbihu , we are looking for the fact that whether, attr->user_data, could be a NULL or not.  And from your answer, it indicates that content of user_data is a pointer to a memory that is pre-allocated and, thus, would not be NULL.  I think the label, user_data, has a different meaning here and that alerts us.  Here, user_data, is actually a field called user_data in the BLE Characteristic.  Usually, in code, label such as, user_data, represents input from user (or application).  It is not the case here.

@sgbihu , do you agree with my explanation?

I don't think there is a need to change anything here.  @eriknyquist , would you agree?